src/app/item-popup/DesktopItemActions.tsx

```diff
@@ -65,6 +69,11 @@ export default function DesktopItemActions({ item }: { item: DimItem }) {
     hideItemPopup();
   };
 
+  const onToggleSidecar = () => {
+    setIsCollapsed(!isCollapsed);
+    dispatch(setSetting('sidecarPreference', !isCollapsed));
```

Comment:
You don't need both state and settings - you can drive this entirely off the setting.